pkg/ctrl/doc.go

```diff
@@ -0,0 +1,186 @@
+
+2. Handle work items by Reconciling user requested state with the actual state in the cluster.
+
+The standard Controller workflow: Watch Event -> Enqueue ObjectKey(s) from Event -> Dequeue ObjectKey -> Reconcile
+
+Controllers often Watch multiple Resource types, such as Pods, ReplicaSets and Deployments, but they should
+enqueue only keys for a single Resource type.  e.g. For Deployment events, enqueue the key of the Deployment.
+For Pod and ReplicaSet Events, enqueue the key of the Deployment that owns the ReplicaSet (directly) or Pod
+(transitively through a ReplicaSet).
+
+Import Concept:
+ObjectKeys are deduplicated when they are enqueued.  1000 Pod Events owned by the same Deployment should
+result in only a single Deployment Reconcile where the Pod Events are mapped to a single Deployment key.  This is why
+Events must be mapped to an ObjectKey of a single type before they are Enqueued instead of when they are Reconciled.
+
+Events may also be transformed into arbitrary sets of ObjectKeys - such as when responding to a cluster resize
+event (add / delete Node) by re-reconciling all instances of a Resource that care about the cluster size.
+
+Reconcile
+
+Reconcile implements the logic to resolve the difference between an object's user defined state
+and the actual cluster state.  This may include reading and writing Kubernetes objects (such as Pods)
+or external objects (such as network attached storage).  A typical Reconcile function may read
+the object for the ObjectKey as well as owned objects, and create or update owned objects.
+
+e.g. if an object declares 5 replicas, and the actual cluster contains only 1 Pod replica, Reconcile
+will create 4 Pods owned by the object.
+
+Reconcile can be implemented as either a type:
+
+	type Reconcile struct {}
+
+	func (Reconcile) Reconcile(ctrl.ObjectKey) (ctrl.ReconcileResult, error) {
+		// Implement business logic of reading and writing objects here
+		return ctrl.ReconcileResult{}, nil
+	}
+
+	controller := &ctrl.Controller{Name: "pod-controller", Reconcile: Reconcile{}}
+
+
+Or as a function:
+
+	controller := &ctrl.Controller{
+	  Name: "pod-controller",
+	  Reconcile: ctrl.ReconcileFunc(func(o ctrl.ObjectKey) (ctrl.ReconcileResult, error) {
+		// Implement business logic of reading and writing objects here
+		return ctrl.ReconcileResult{}, nil
+	  })
+	}
+
+
+Watchables and EventHandlers
+
+A Controller is configured to Watch for Events by calling the Watch function.  The Watch function
+accepts a Watchable source of Events and an EventHandler that enqueues work for Events.
+
+	// Watch for Pod Events and enqueue the key of the Pod
+	controller.Watch(
+		ctrl.Resource{Group: "core", Version: "v1", Resource: "pods"},
+		ctrl.Enqueue{},
+	)
+
+Resource
+
+Resource implements Watchable and contains the Group, Version, and Resource of an API to Watch for Events.
+
+	ctrl.Resource{Group: "core", Version: "v1", Resource: "pods"}
+
+Channel
```

Comment:
You haven't really introduced channel before that, and I honestly don't understand how they would be useful by reading the description below?